### PR TITLE
Document maintanance of tc39

### PR DESCRIPTION
**Pull request by olegbespalov**
## What?

This documents my recent learnings from the conversation with @mstoykov and recent PR about maintaining of tc39 test

## Why?

Knowledge sharing

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`make lint`) and all checks pass.
- [ ] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: js/tc39/README.md [hunk 1]
```diff
@@ -7,15 +7,23 @@ Ways to use it:
...
+* Due to changes to Sobek it's common for the error to change, or there to be now a new error on the previously passing test, or (hopefully) a test that was not passing but now is.
+In all of those cases `breaking_test_errors-*.json` needs to be updated. Run the test with `-update` flag to update: `go test -update`.
+* Important to mention that there should be a balance between just updating the list of the errors and actually fixing errors. So it's recommended to check the diff of the `breaking_test_errors-*.json` and case by case decide if the error should be updated or the test should be fixed.
+
+> [!NOTE]
+> some text editors/IDEs will try to parse files ending in `json` as JSON, which given the size of `breaking_test_errors-*.json` might be a problem when it's not actually a JSON (before the edit). So it might be a better idea to name it something different if editing by hand and fix it later.
```
Comment (mstoykov) [inline js/tc39/README.md:26]: this note is kind of not relevant anymore as the previous explanations was based on you manually removing stuff from the json file after redirected `go test` output to it. 

As this no longer happens I think we can drop this.

```suggestion
```

File: js/tc39/README.md [hunk 2]
```diff
@@ -7,15 +7,23 @@ Ways to use it:
...
+The full list of failing tests, and the error, is in `breaking_test_errors-*.json`. All errors list there with the corresponding error will *not* be counted as errors - this is what the test expects, those specific errors. See reasons for this at the end of this document.
 
 This is a modified version of [the code in the original goja
 repo](https://github.com/dop251/goja/blob/master/tc39_test.go) that Sobek was forked from.
 
+## Maintaining the tests
+
+There are few things to keep in mind when maintaining the tests.
+
+* For the most of the version we aim to have the same version of the test suite as the one that [uses Sobek](https://github.com/grafana/sobek/blob/main/.tc39_test262_checkout.sh#L3). So if the Sobek brings a new version of the test suite, we should update the version of the test suite in this package as well.
```
Comment (joanlopez) [inline js/tc39/README.md:19]: The beginning of the sentence is a little confusing, I think

Comment (olegbespalov) [reply js/tc39/README.md:19]: Right, was trying to write this down in-between meetings :sweat_smile: 

```suggestion
* For the most time, we aim to have the same version (commit hash) of the test suite as the one that [uses Sobek](https://github.com/grafana/sobek/blob/main/.tc39_test262_checkout.sh#L3). So if Sobek brings a new version of the test suite, we should update the version of the test suite in this package as well.
```

**Review by mstoykov — APPROVED** [review]

File: js/tc39/README.md (hunk 2 above)
Comment (olegbespalov) [reply js/tc39/README.md:19]: @joanlopez, how about this version?

Comment (joanlopez) [reply js/tc39/README.md:19]: Yeah, now looks better. Thanks! 🙇🏻

**Review by mstoykov — APPROVED** [review]

**Review by joanlopez — APPROVED** [review]

**Pull request merged**